src/test/regress/expected/multi_alter_table_add_constraints.out

```diff
@@ -0,0 +1,501 @@
+DETAIL:  Failing row contains (1, product_1, 2, 3).
+CONTEXT:  while executing command on localhost:57638
+DROP TABLE products;
+-- Check "CHECK CONSTRAINT" with reference table 
+CREATE TABLE products_ref (
+    product_no integer,
+    name text,
+    price numeric,
+    discounted_price numeric
+);
+SELECT create_reference_table('products_ref');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+-- Can add column and table constraints
+ALTER TABLE products_ref ADD CONSTRAINT p_check CHECK(price > 0);
+ALTER TABLE products_ref ADD CONSTRAINT p_multi_check CHECK(price > discounted_price);
+-- Error out. Since the last one conflicting with p_multi_check.
+INSERT INTO products_ref VALUES(1, 'product_1', -1, -2);
+ERROR:  new row for relation "products_ref_1400100" violates check constraint "p_check_1400100"
+DETAIL:  Failing row contains (1, product_1, -1, -2).
+CONTEXT:  while executing command on localhost:57637
+INSERT INTO products_ref VALUES(1, 'product_1', 5, 3);
+INSERT INTO products_ref VALUES(1, 'product_1', 2, 3);
+ERROR:  new row for relation "products_ref_1400100" violates check constraint "p_multi_check_1400100"
+DETAIL:  Failing row contains (1, product_1, 2, 3).
+CONTEXT:  while executing command on localhost:57637
+DROP TABLE products_ref;
+-- Check "CHECK CONSTRAINT" with append table
+CREATE TABLE products_append (
+    product_no int,
+    name varchar(20),
+    price int,
+    discounted_price int
+);
+SELECT create_distributed_table('products_append', 'product_no', 'append');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- Can add column and table constraints
+ALTER TABLE products_append ADD CONSTRAINT p_check CHECK(price > 0);
+ALTER TABLE products_append ADD CONSTRAINT p_multi_check CHECK(price > discounted_price);
+-- Error out,since the third row conflicting with the p_multi_check
+COPY products_append FROM STDIN DELIMITER AS ',';
+ERROR:  new row for relation "products_append_1400101" violates check constraint "p_multi_check"
+DETAIL:  Failing row contains (1,  Product_3, 8, 10).
+DROP TABLE products_append;
+-- Check "EXCLUSION CONSTRAINT"
+CREATE TABLE products (
+    product_no integer,
+    name text,
+    price numeric
+);
+SELECT create_distributed_table('products', 'product_no');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- Can only add exclusion constraint on distribution column (or group of columns
+-- including distribution column)
+-- Command below should error out since 'name' is not a distribution column
+ALTER TABLE products ADD CONSTRAINT exc_name EXCLUDE USING btree (name with =); 
+ERROR:  cannot create constraint on "products"
+DETAIL:  Distributed relations cannot have UNIQUE, EXCLUDE, or PRIMARY KEY constraints that do not include the partition column (with an equality operator if EXCLUDE).
+-- We can add composite exclusion because none of pair of rows are conflicting
```

Comment:
We can add composite exclusion -> corrent
because none of pair of rows are conflicting -> did not understand that